crates/red_knot_python_semantic/resources/mdtest/literal/collections/list.md

```diff
@@ -6,3 +6,33 @@
 x = []
 reveal_type(x)  # revealed: list
 ```
+
+## Indexing into lists
+
+A list can be indexed into with:
+- numbers
+- slices
+
+
+```py
+x = [1, 2, 3]
+reveal_type(x)  # revealed: list
+# TODO reveal int
+reveal_type(x[0])  # revealed: @Todo
+# TODO reveal list
+reveal_type(x[0:1])  # revealed: @Todo
+# TODO error
+reveal_type(x["a"])  # revealed: @Todo
+```
+
+## Assignments within list assignment
+In assignment, we might also have a named assignment.
+This should also get type checked.
+```py
+x = [1, 2, 3]
+x[0 if (y := 2) else 1] = 5
+# TODO error? (indeterminite index type)
+x["a" if (y := 2) else 1] = 6
+# TODO error (can't index via string)
+x["a" if (y := 2) else "b"] = 6
+```
```

Comment:
I just realized that this was in `literal/collections/list.md` and not `collections/list.md`. Please advise on whether this should be extracted to be placed elsewhere

Reply:
I would put these tests in `subscript/list.md`.